src/Uno.UI/UI/Xaml/Controls/Image/Image.wasm.cs

```diff
@@ -88,36 +88,38 @@ private ExceptionRoutedEventArgs ImageFailedConverter(object sender, string e)
 
 			if (e.NewValue is ImageSource source)
 			{
-				_sourceDisposable.Disposable = source.Subscribe(img =>
+				void OnSourceOpened(ImageData img)
 				{
 					switch (img.Kind)
 					{
 						case ImageDataKind.Empty:
 							_htmlImage.SetAttribute("src", "");
 							break;
 
-						case ImageDataKind.Base64:
+						case ImageDataKind.DataUri:
 						case ImageDataKind.Url:
 						default:
 							if (MonochromeColor != null)
 							{
-								WebAssemblyRuntime.InvokeJS("Uno.UI.WindowManager.current.setImageAsMonochrome("
-									+ _htmlImage.HtmlId + ", \""
-									+ img.Value + "\", \""
-									+ MonochromeColor.Value.ToHexString() + "\");");
+								WebAssemblyRuntime.InvokeJS("Uno.UI.WindowManager.current.setImageAsMonochrome(" + _htmlImage.HtmlId + ", \"" + img.Value + "\", \"" + MonochromeColor.Value.ToHexString() + "\");");
 							}
 							else
 							{
+								Console.WriteLine($"Setting img src to {img.Value}");
 								_htmlImage.SetAttribute("src", img.Value);
 							}
+
 							break;
 
 						case ImageDataKind.Error:
 							_htmlImage.SetAttribute("src", "");
-							_htmlImage.InternalDispatchEvent("error", EventArgs.Empty);
+							var errorArgs = new ExceptionRoutedEventArgs(this, img.Error?.ToString());
+							_htmlImage.InternalDispatchEvent("error", errorArgs);
 							break;
 					}
-				});
+				}
+
+				_sourceDisposable.Disposable = source.Subscribe(OnSourceOpened);
```

Comment:
most likely needs to be set to null before
```suggestion
				_sourceDisposable.Disposable = null;
				_sourceDisposable.Disposable = source.Subscribe(OnSourceOpened);
```